File: search/linear/linear-search.py
def linear_search(arr=[1,2,3,4,5],find=5) -> int:
    position = None
    if arr and find:
      for index in range(0,len(arr)):
            if arr[index] == find:
                position = index
                break
    return position

if __name__ == "__main__":
    array = []
    while True:
        temp = input("Enter a value into the array (press enter to finish): \n")
        if temp:
            array.append(temp)
        else:
            break
    value = input("What value would you like to find: \n")
    res = linear_search(array,value)
    print(res if res != None else "Not found")

File: search/linear/linearSearch.go
package main

import "fmt"

func linear_search(arr []string, target string) int {
	index := -1
out:
	for i := 0; i < len(arr); i++ {
		if arr[i] == target {
			index = i
			break out
		}
	}
	return index
}

func main() {
	var arr []string
	var its int
	var targ string
	fmt.Println("enter the amount of values you wish to enter")
	fmt.Scan(&its)
	for i := 0; i < its; i++ {
		temp := ""
		fmt.Println("enter a value if you are finished press enter")
		fmt.Scan(&temp)
		arr = append(arr, temp)

	}
	fmt.Println("enter the value you ant to find")
	fmt.Scan(&targ)

	fmt.Printf(" target: %v \n index found: %v \n", targ, linear_search(arr, targ))
}
